### PR TITLE
@fredmk's axios refactor rebased on dev

### PR DESCRIPTION
**Pull request by bewest**
I have rebased the axios branch on top of dev with these results.
https://github.com/FredMK/minimed-connect-to-nightscout/tree/axios

### CODE REVIEW

**Comment by sulkaharo** [comment]
Great! Do we need a branch of Nightscout that points to this to enable testing?

**Comment by bewest** [comment]
Yes, I've been trying to test it a little bit more since there are substantial changes, but short on time.  If we think a bounty/contract could solve this for reasonable price, T1Pal might be willing to fund development on this.

**Comment by bewest** [comment]
Tying it all together:
* https://github.com/nightscout/minimed-connect-to-nightscout/issues/5#issue-687862738
* https://github.com/nightscout/minimed-connect-to-nightscout/issues/9

File: carelink.js [hunk 1]
```diff
@@ -2,13 +2,16 @@
...
+    axios = require('axios').default,
+    axiosCookieJarSupport = require('axios-cookiejar-support').default,
+    tough = require('tough-cookie'),
     urllib = require('url'),
-    request = require('request');
+    qs = require('qs');
 
 var logger = require('./logger');
 
+process.env["NODE_TLS_REJECT_UNAUTHORIZED"] = 0;
```
Comment (bewest) [inline carelink.js:13]: Nervous about shipping this to production.  We would like to validate that we are talking to Medtronic without someone tampering.

**Comment by bewest** [comment]
@FredMK, your changes are merged into `dev` now with tests passing.

File: carelink.js (hunk 1 above)
Comment (FredMK) [reply carelink.js:13]: the current code not using this

**Pull request merged**